openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java

```diff
@@ -276,12 +283,20 @@ private static RecordFieldWithDefaultValueNode getRecordFieldWithDefaultValueNod
      * @param fieldSchema Schema for data type
      * @return {@link MetadataNode}
      */
-    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema) {
+    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema)
+            throws BallerinaOpenApiException {
         if (isConstraintAllowed(typeName, fieldSchema)) {
-            if (GeneratorUtils.isStringSchema(fieldSchema)) {
+            String ballerinaType = convertOpenAPITypeToBallerina(fieldSchema);
+            //The current tool still supports byte and binary formats in string types.
+            //The rest of the string formats are mapped to the string type. Therefore, we do not provide constraint
+            //support for byte and binary formats.
+            //TODO: Please verify if constraints are supported when attempting to add new string format type support.
```

Comment:
Do we need this todo? :) IINM such cases will be automatically handled as now we validate the counterpart ballerina type as well, before adding the annotation. WDYT?

Reply:
Yes. With this new approach we do not need to worry about the OpenAPI formats since we generate constraints based on the type of the field anyway. If that type is not supported in constraints then that will not be generated.

Reply:
We included this comment for future reference, in case someone needs to address the remaining string format types. That person may benefit from the context of the constraint. 🙂

Reply:
The comment added here seems the better place to warn in such case https://github.com/ballerina-platform/openapi-tools/blob/8f88e4116a61a52b2101fada200944fa93e086dc/openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorConstants.java#L317

Reply:
@aneeshafedo yes this is the place we should have the comment ideally